rdrf/rdrf/services/io/actions/patient_status_report.py

```diff
@@ -393,12 +393,16 @@ def _completed(self, patient_model, form_name, data, percentage=False):
 
     def _get_patients(self):
         user_working_groups = self.user.working_groups.all()
-
-        for patient_model in Patient.objects.filter(rdrf_registry__code__in=[self.registry_model.code],
-                                                    working_groups__in=user_working_groups):
-
-            if self.has_no_filter or self._include_patient(patient_model):
-                yield patient_model
+        if not self.has_filter:
+            return Patient.objects.filter(rdrf_registry__code__in=[self.registry_model.code],
+                                          working_groups__in=user_working_groups).iterator()
+        else:
+            def patient_iterator():
+                for patient_model in Patient.objects.filter(rdrf_registry__code__in=[self.registry_model.code],
+                                                            working_groups__in=user_working_groups):
+                    if self._include_patient(patient_model):
```

Comment:
if we can push out the test for self.start_date and end_date outside the loop would be good